server/models/image.py

```diff
@@ -0,0 +1,80 @@
+from girder.models.item import Item
+from girder.models.model_base import ValidationException
+
+from .meta import TCGAModel
+
+
+class Image(TCGAModel, Item):
+
+    TCGAType = 'image'
+
+    def validate(self, doc, **kwargs):
+        super(Image, self).validate(doc, **kwargs)
+        if 'largeImage' not in doc:
+            raise ValidationException(
+                'An image item must be a "large_image"'
+            )
+        slide = self.model('slide', 'digital_slide_archive').load(
+            doc['folderId'], force=True)
+        if not slide or self.getTCGAType(slide) != 'slide':
+            raise ValidationException(
+                'An image must be a child of a slide'
+            )
+        tcga = self.getTCGA(doc)
+        if not self.case_re.match(tcga.get('case', '')):
+            raise ValidationException(
+                'Invalid case name in TCGA metadata'
+            )
+        if not self.barcode_re.match(tcga.get('barcode', '')):
+            raise ValidationException(
+                'Invalid barcode in TCGA metadata'
+            )
+        if not self.uuid_re.match(tcga.get('uuid', '')):
+            raise ValidationException(
+                'Invalid uuid in TCGA metadata'
+            )
+
+        return doc
+
+    def _setLargeImage(self, doc, fileId, user, token):
+        if doc.get('largeImage', {}).get('fileId') == fileId:
+            return
+        file = self.model('file').load(fileId, user=user)
+        try:
+            return self.model('image_item', 'large_image').createImageItem(
+                doc, file,
+                user=user, token=token
+            )
+        except Exception:
+            raise ValidationException(
+                'Could not generate a large_image item'
+            )
+
+    def _findImageFile(self, doc):
+        for file in self.childFiles(doc):
+            if self.image_re.match(file['name']):
+                return file['_id']
+
+    def importDocument(self, doc, **kwargs):
+        """Import a slide item into a `case` folder."""
+        user = kwargs.get('user')
+        token = kwargs.get('token')
+        fileId = self._findImageFile(doc)
+        if fileId is None:
+            raise ValidationException(
+                'Could not find a TCGA slide in item'
+            )
+        self._setLargeImage(doc, fileId, user, token)
```

Comment:
If large_image has to create a job, it is still possible that it will fail to make the file referenced by `fileId` into a largeImage (since the conversion could fail).  One option would be to use `createJob=False` as pat of the `createLargeImage` call, which would prevent this from being an asynchronous process.  Otherwise, at least at a TODO note that this should be validated in some way.